src/mappings/index.ts

```diff
@@ -288,9 +288,11 @@ export async function handleOfferPlace(context: Context): Promise<void> {
   // logger.debug(`offer: ${JSON.stringify({ ...event, price: String(event.amount), expiresAt: String(event.expiresAt)  }, null, 2)}`)
   const id = createTokenId(event.collectionId, event.sn);
   const entity = ensure<NE>(await get(context.store, NE, id));
-  plsBe(real, entity);
+  
+  // entity doesn't need to exist
```

Comment:
NFT has to exist

Reply:
https://github.com/galacticcouncil/Basilisk-node/blob/master/pallets/marketplace/src/tests/make_offer.rs#L6

Reply:
& https://github.com/galacticcouncil/Basilisk-node/blob/master/pallets/marketplace/src/lib.rs#L169
no check if NFT exist

Reply:
Good catch!

But It can fail on unknown foreign key in DB 🤔

Reply:
> Good catch!
> 
> But It can fail on unknown foreign key in DB 🤔

which foreign key? it only concerns Offers, if you find a particular case I would like to see it

I think we have no choice here you can check  https://github.com/kodadot/snek/issues/127 to know why I've made this particular change

Reply:
from Offers to NFTs 
But if you try that and it does not fail I am happy to try that in rococo